src/pages/api/consent-cookie/index.ts

```diff
@@ -0,0 +1,55 @@
+import { Endpoint, HttpStatus, setServerSideCookies } from "@fwl/web";
+import { wrapMiddlewares, Middleware } from "@fwl/web/dist/middlewares";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { Handler } from "@src/@types/handler";
+import { logger } from "@src/configs/logger";
+import getTracer from "@src/configs/tracer";
+import { ERRORS_DATA, webErrorFactory } from "@src/errors/web-errors";
+import { noAuthBasicMiddlewares } from "@src/middlewares/basic-middlewares";
+
+const getHandler: Handler = (request, response): Promise<void> => {
+  return getTracer().span("get-consent-cookie handler", async (_span) => {
+    const consentCookie = request.cookies["user-consent"];
+    const consentCookieResponse = {
+      isSet: consentCookie ? true : false,
+      content: consentCookie ? consentCookie : null,
+    };
+    response.statusCode = HttpStatus.OK;
+    response.setHeader("Content-Type", "application/json");
+    return response.end(JSON.stringify(consentCookieResponse));
+  });
```

Comment:
Couple things 😄.

Don't you see any value on adding `consentCookieResponse ` as a span attribute?

Should we have a `post` and a `patch`?

Also, reopening the subject regarding the `return` policy. We said that we wanted de have the `return` explicitly on its own line at the end of the handler. What do you think about it now? We previously said that having it in isolation would help no forgetting it, but it's not pretty. I'm not sure what to think anymore.

Reply:
It could be useful to add it in a span attribute indeed. I will do it 🙂 

For the `post`/`patch` thing, I'm not sure what's best. I followed what we already have for another cookie (the locale one) and there isn't a `post` handler neither. I'm pretty sure that the browser will override a cookie with the same name when trying to set it, so technically we could stay this way, this handler could be used for both setting it for the 1st time and to modify it if it is already present. Wdyt ? 

Regarding the return policy, the only reason I did it this way is that I found that it prevents and solves the issue with the console error "API resolved without a response..." that we have in local env for a few days. I'm still OK with what we talked about having the return on its own line even when it is apparently not needed to have one.

Reply:
👍 

Well, if it override the value each time, it should be a post then I think 🤔 (even if I was wrong with the locale one haha). I'm not against leaving it as it is for now, and change both locale and this one in the future if we decide to do so :).

Ok great. We'll see what comes back from my task to fix all the API return statements.

Reply:
@Fenntasy maybe you can answer our interrogations regarding this cookie thing behaviour. When the browser must set a cookie, and that a cookie with the same name already exists, what really happens ? Does it erase the old one and set the new one, or does it update it ?

Reply:
> When the browser must set a cookie, and that a cookie with the same name already exists, what really happens ? Does it erase the old one and set the new one, or does it update it ?

what's the difference between erase/create and update? 🤔 I see a technical difference but I don't see the impact. If I had to guess, I'd say erase/create though.

However, we got bit on sparta when dealing with cookies, you can't set them in a `fetch` call but you need to navigate to a page somewhere, old style. Otherwise it won't work

Reply:
In fact, there's no difference between the two, but we were wondering what kind of HTTP verb between `POST` and `PATCH` we should use for this handler. We don't have any issue with it, it works just fine, and the naming of the handler action won't change anything technically but we wanted to be the most accurate regarding HTTP verbs usage 😅